darshan-runtime,darshan-util: new version released + move to github

shanedsnyder opened the pull request:
- add 3.3.1 tag for runtime/util components
- remove preferred tag to make sure we use our most recent release
- add url_for_version() for each package to pick right URL based on version number (pre 3.3.1 versions not available on GitHub)

Commit:
darshan runtime/util new version, move to github

- add 3.3.1 tag for runtime/util components
- remove preferred tag to make sure we use our most recent release
- add url_for_version() for each package to pick right URL based
  on version number (pre 3.3.1 versions not available on GitHub)

alalazo reviewed (changes requested):
Due to a style check introduced in #23947 there's need to tweak the import order in this PR:
```python
import os

from spack import *
```

Commit:
fix import order for darshan-runtime

Commit:
update darshan git references for github

shanedsnyder commented:
Should have the import order fixed, as well as a small update to the `git` field in each package and a change from master->main branches -- last 2 were needed to complete transition from gitlab to github.

Commit:
forgot new line to appease style reqs for import

carns reviewed (dismissed):
looks good to me

Commit:
revert change to github release tarballs

GitHub releases do not account for git submodules that are used
by Darshan, so we will continue using our own FTP server

shanedsnyder commented:
Sorry for the late commits here...we just realized GitHub doesn't support all we need to use it for managing our releases, so we will continue to maintain our releases on our own FTP server. Most recent commit redirected the `url` back to our FTP server and also recalculated the version checksum for there.

I think I'm actually done now :)
